Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp

```diff
@@ -314,6 +319,18 @@ namespace AzFramework
             // previously-spawned entities from a previous SpawnEntities or SpawnAllEntities call.
             InitializeEntityIdMappings(entitiesToSpawn, ticket.m_entityIdReferenceMap, ticket.m_previouslySpawned);
 
+            for (const auto& nameSpawnablePair : ticket.m_spawnable->GetDependentSpawnableList())
+            {
+                const AZ::Name& controllerName = nameSpawnablePair.first;
+                const AZ::Data::Asset<AzFramework::Spawnable>& dependentSpawnableAsset = nameSpawnablePair.second;
+
+                auto controllerIt = m_dependentSpawnableControllers.find(controllerName);
+                if (controllerIt != m_dependentSpawnableControllers.end())
+                {
+                    controllerIt->second->ProcessSpawnable(*dependentSpawnableAsset, ticket.m_entityIdReferenceMap, request.m_serializeContext);
+                }
+            }
```

Comment:
The idea of having dependent spawnables is a good one, especially considering the Networking versions are in a way children of the main Prefab. I do have a few reservations with this approach though:

1. This is limited to just the SpawnAllEntities call, so doesn't work when individual entities are spawned.
2. The ticket doesn't actually own the entities spawned from the dependent spawnable with makes the ownership nebulous because for any spawned group of entities some will be tracked by the ticket and some won't. This also would cause issues when iterating over the entities or claiming them.
3. There ends up being quite a lot of duplicated code in the implementation of the controller.
4. There doesn't seem to be anything to delete the entities that are spawned from the dependent.
5. Reloading Prefabs wouldn't cause the 

As an alternative would the following work:

- Extend the ticket with support for adding additional spawnables? 
  - There would be an additional function to add additional spawnables.
  - The internal bookkeeping would need to keep track of what spawnable created which entity. This can just be an additional index into the array of spawnables on the ticket as additional spawnables would be removable.
  - SpawnAllEntities would spawn all entities including in the additional spawnables in the order the additional spawnables were added.
  - SpawnEntities can be done in a few ways, but my current thinking is keeping it simple and keep counting, so if the main spawnable has 10 entities then requesting entity 12 means number 2 in the first additional spawnable.
  - The callback for ListIndicesAndEntities would need to be extended to also include the assets.
  - ReloadSpawnable would be limited to reloading just the provided spawnable which can now be an additional spawnable as well.
  - The rest of the functions would continue to function as is.
- The spawnable meta data could be extended to store asset ids as well. This would avoid having to add more the API and storing another `AZStd::vector` but would require the stored name is known before hand to retrieve the asset, but from the PR is looks like that would be possible for network spawnables.
- The additional spawnable never seems to be loaded in this PR, but I think there needs to be some code that handles this. In particular, if the additional spawnable is setup as a dependency in the builder than Asset Streaming will load the additional spawnable but if there's nothing in the main spawnable that holds on to the additional spawnable (such as patching up the spawnable asset in the meta data) then the additional spawnable will be unloaded and reloaded at a later point.